### ISSUE TITLE
screenreader support

### ISSUE THREAD

**Issue by Veyfeyken** [Selz/plyr#103]
The current version does not work with the most popular screenreaders.

VoiceOver doesn't read the labels for the checkboxes (toggle mute and captions).
NVDA and Jaws only read the play-button because it has an aria-label. The other controls are read as "button".

Every control has an text-alternative `<span class="player-tooltip">` that's shown on hover or on focus.

I think they're being ignored because they're set to "visibility:hidden". 
Using CSS "clip" like you do in the class "sr-only" might fix it.

Using `<span>`'s with a description of the control might be better than using aria-labels. Aria support is not reliable among screenreaders yet.

**Comment by SamPotts** [comment]
Ah sorry to hear about this. I'll install NVDA on my Windows box and fix it up. I'd been testing on Voiceover on Mac which I guess is fairly niche but even that doesn't read the tooltips for the checkboxes even though they have labels but I think it's to do with `visibility: hidden

**Comment by SamPotts** [comment]
What did you mean by "Using 's with a description"?

**Comment by Veyfeyken** [comment]
`<span>`'s

**Comment by Veyfeyken** [comment]
I think using the current descriptions that are shown on hover is the best approach. Don't use ARIA if you don't have to.
http://www.w3.org/TR/aria-in-html/#first-rule-of-aria-use

**Comment by SamPotts** [comment]
Ok awesome. I'll work on this a bit later today/tomorrow as it should be a quick win. Thanks for reporting it :+1:

**Comment by Veyfeyken** [comment]
Thx for looking into this. I hope I can recommend the player as fully accessible in the near future.

**Comment by SamPotts** [comment]
Hey, can you check out http://plyr.io? I think I've fixed this now?

**Comment by Veyfeyken** [comment]
* Works with NVDA 2015.2 in Firefox 
* Works with Jaws 16 and 15 in IE11. 
* Unfortunately VoiceOver still ignores the `<label>`'s for toggle mute and captions. (OS X Yosemite 10.10.4 with Safari 8.0.7). You're hiding them with opacity: 0, that shouldn't impact VoiceOver. I can't explain this behaviour.

**Comment by MichielBijl** [comment]
Why does the mute toggle button use an `<input type="checkbox">`? Why not use a `<button>`? You could mark it up as [described in the Authoring Practices](https://rawgit.com/w3c/aria/master/practices/examples/button/button.html).

This could potentially solve the issue; as none of the other controls (`buttons`) seem to have the problem.

That only leaves the subtitle button, which can use the same—toggle button—markup as the mute button for now. If you ever decide to add multiple subtitle support, this needs other markup (because it not only toggles, but selects a specific subtitle).

**Comment by SamPotts** [comment]
Good suggestion @MichielBijl , I'll change them to toggle buttons which should resolve it for Safari. Oddly Chrome on OSX reads the labels correctly.

**Comment by SamPotts** [comment]
I've implemented the 'aria-pressed' attribute on toggle buttons and it seems to get picked up in Chrome on OSX as a toggle button (also reads selected state) using voiceover. NVDA in windows also works too and states they are toggle buttons. 

I've also made some improvements so that captions are read by NVDA etc (using live regions). 

If you find anything else please let me know. I'm keen to make sure everything is top notch a11y wise.

**Comment by MichielBijl** [comment]
Nice work!

**Comment by Veyfeyken** [comment]
Hi, looks like all controls are read by most screenreader-browser-combinations now. Good job. 

Unfortunately "aria-pressed" is not well supported. Using toggle buttons like the play-pauze button would be better. "Mute - unmute", "captions on - captions off". That 'll work for every screenreader in any context. Again, if you can do it without ARIA, that's always better.

Regarding captions with live regions. This is more of a problem then an enhancement. You're using aria-live="assertive" and captions are on by default. The "assertive" property interrupts screenreaders, regardless of what they're doing. Therefore you're forcing screenreader users to listen to the captions. Captions are primarily meant for people with hearing problems, not visually impaired users. I would recommend undoing this.

**Comment by MichielBijl** [comment]
@Veyfeyken do you have any sources for aria-pressed support? According to [this post on TPG blog](http://www.paciellogroup.com/blog/2010/10/jaws-support-for-aria/) JAWS supports it. Here is a document that states [it's supported as far back as JAWS 8 + IE8](http://files.paciellogroup.com/blogmisc/ARIA/togglebutton.html). It was also [added to Bootstrap](https://github.com/twbs/bootstrap/commit/e1d40f7633886298a9fb417ed4f118844ed83a0d) by @patrickhlauke.

If there any tests for this that I failed to find, could you post them here?

**Comment by SamPotts** [comment]
Hey @Veyfeyken, I've removed the live region now so captions should no longer be read. I thought the same after I'd released it that perhaps it wasn't really needed given the captions intended purpose and thought about removing that. :+1:

**Comment by SamPotts** [comment]
Happy to make a button change at some point...

**Comment by Veyfeyken** [comment]
@MichielBijl , our own tests. VoiceOver ignored it. Jaws and NVDA have support. My colleague tested with Supernova, popular in Belgium and the Netherlands. To my surprise, even Supernova supports it. So only VoiceOver (with Safari) is a problem. So I humbly take back my statement, support is better then I thought. We didn't test with older browser versions or mobile.

**Comment by MichielBijl** [comment]
@Veyfeyken thanks for the information :)

Here is a link to [my conversation on a11ySlackers](https://gitter.im/w3c/a11ySlackers?at=55c9e75c7f578b465ff4ac3d).

**Comment by SamPotts** [comment]
Chrome and Opera read the buttons as toggle buttons in Voiceover. I guess I should be consistent and change the play/pause to be a toggle switch or do you think it's ok as is? 

Thanks again for your help :+1:

**Comment by MichielBijl** [comment]
@SamPotts the play button label changes to “Pause”; it's not a toggle button. So that is fine as is.
